tests/functional/remote/09-restart-running-file-install.t

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Test restart remote init, file install and task messaging works for running tasks
+# The test works as follows:
+# - Starts workflow off with a remote starter task which will not complete
+#   until restarted. It greps for updated file which will be remote installed on
+#   restart.
+# - On start of this starter task, update the file that is included in
+#   remote file install for starter task.
+# - Stop the workflow - this will leave starter task orphaned
+# - Restart, starter task should remote install, authentication keys will be
+#   updated which is verified by checking task messaging works (checks for
+#   "(received)succeeded" in the logs).
+
+export REQUIRE_PLATFORM='loc:remote comms:tcp'
+. "$(dirname "$0")/test_header"
+set_test_number 6
+
+init_workflow "${TEST_NAME_BASE}" <<'__FLOW_CONFIG__'
+[scheduler]
+    install = changing-file
+    [[events]]
+        abort on inactivity timeout = True
+        inactivity timeout = PT1M
+[scheduling]
+    [[graph]]
+        R1 = """
+        starter:start => file-changer => stopper
+        """
+[runtime]
+    [[starter]]
+        platform = ${CYLC_TEST_PLATFORM}
+        script = """
+            while ! grep 'Restart Play' "${CYLC_WORKFLOW_RUN_DIR}/changing-file"; do
+                sleep 1
+            done
+        """
+    [[stopper]]
+        script = cylc stop --now "${CYLC_WORKFLOW_ID}"
+    [[file-changer]]
+        script = echo Restart Play > ${CYLC_WORKFLOW_RUN_DIR}/changing-file
```

Comment:
Because this runs before `stopper` I think `starter` *can* succeed *before* the workflow is shut down?

Maybe `starter` could grep for `DONE` in the workflow log (removes the need for `file-changer`)?

Reply:
Re-looking at this I'm not sure it can, unless I'm misunderstanding (very possible)?

- `starter` is a remote task that cannot complete until the restart (triggered by `stopper` and the test) has taken place. 

This is because the message in the file it is grepping for will not be there until the `file-changer`, which is a local task (it won't change the file on the remote), changes the message, and reinstallation takes place. 

- This leaves `starter` as a running task.
- Only after remote init has completed (placing the updated `changing-file`) on restart will `starter` succeed. At which point test complete.

What do you think?

Reply:
Ah, heck, it's using file-install to sync the file to the remote, gotcha.

You've put that in the comments above which I completely skipped.